docs/usage.rst

```diff
@@ -0,0 +1,134 @@
+
+.. code-block:: bash
+
+   spack install pika
+
+See
+
+.. code-block:: bash
+
+   spack info pika
+
+for available options.
+
+Manual installation
+-------------------
+
+If you'd like to build pika manually you will need CMake 3.22.0 or greater and a recent C++ compiler
+supporting C++17:
+
+- `GCC <https://gcc.gnu.org>`_ 9 or greater
+- `clang <https://clang.llvm.org>`_ 11 or greater
+
+Additionally, pika depends on:
+
+- `header-only Boost <https://boost.org>`_ 1.71.0 or greater
+- `hwloc <https://www-lb.open-mpi.org/projects/hwloc/>`_ 1.11.5 or greater
+- `fmt <https://fmt.dev/latest/index.html>`_ 9.0.0 or greater
+
+pika optionally depends on:
+
+* `gperftools/tcmalloc <https://github.com/gperftools/gperftools>`_, `jemalloc
+  <http://jemalloc.net/>`_, or `mimalloc <https://github.com/microsoft/mimalloc>`_. It is *highly*
+  recommended to use one of these allocators as they perform significantly better than the system
+  allocators. You can set the allocator through the CMake variable ``PIKA_WITH_MALLOC``. If you want
+  to use the system allocator (e.g. for debugging) you can do so by setting
+  ``PIKA_WITH_MALLOC=system``.
+* `CUDA <https://docs.nvidia.com/cuda/>`_ 11.0 or greater. CUDA support can be enabled with
+  ``PIKA_WITH_CUDA=ON``. pika can also be built with nvc++ from the `NVIDIA HPC SDK
+  <https://developer.nvidia.com/hpc-sdk>`_. In the latter case, set ``CMAKE_CXX_COMPILER`` to
+  ``nvc++``.
+* `HIP <https://rocmdocs.amd.com/en/latest/index.html>`_ 5.2.0 or greater. HIP support can be
+  enabled with ``PIKA_WITH_HIP=ON``.
+* `MPI <https://www.mpi-forum.org/>`_. MPI support can be enabled with ``PIKA_WITH_MPI=ON``.
+* `Boost.Context <https://boost.org>`_ on macOS or exotic platforms which are not supported by the
+  default user-level thread implementations in pika. This can be enabled with
+  ``PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON`` (TODO rename option...).
+* `stdexec <https://github.com/NVIDIA/stdexec>`_. stdexec support can be enabled with
+  ``PIKA_WITH_STDEXEC=ON`` (currently tested with commit `7a47a4aa411c1ca9adfcb152c28cc3dd7b156b4d
+  <https://github.com/NVIDIA/stdexec/commit/7a47a4aa411c1ca9adfcb152c28cc3dd7b156b4d>`_).  The
+  integration is experimental. See :ref:`pika_stdexec` for more information about the integration.
+
+If you are using `nix <https://nixos.org>`_ you can also use the ``shell.nix`` file provided at the root of the repository to quickly enter a development environment:
+
+.. code-block:: bash
+
+   nix-shell <pika-root>/shell.nix
+
+The ``nixpgks`` version is not pinned and may break occasionally.
+
+Including in CMake projects
+---------------------------
+
+Once installed, pika can be used in a CMake project through the ``pika::pika`` target. Other ways of
+depending on pika are likely to work but are not officially supported.
+
+.. code-block:: cmake
+
+   find_package(pika REQUIRED)
+   add_executable(app main.cpp)
+   target_link_libraries(app PRIVATE pika::pika)
```

Comment:
```suggestion
Once installed, pika can be used in a CMake project through the ``pika::pika`` target:

.. code-block:: cmake

   find_package(pika REQUIRED)
   add_executable(app main.cpp)
   target_link_libraries(app PRIVATE pika::pika)
   
Other ways of depending on pika are likely to work but are not officially supported.
```

Move the "Other ways of depending on pika [...]" sentence after the code block?

Reply:
Yep, good idea!